packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -136,7 +136,10 @@ abstract class Route<T> {
   @mustCallSuper
   TickerFuture didPush() {
     return TickerFuture.complete()..then<void>((void _) {
-      navigator.focusScopeNode.requestFocus();
+      final Offstage offstageWidget = navigator.context.findAncestorWidgetOfExactType<Offstage>();
+      if (offstageWidget == null || !offstageWidget.offstage) {
```

Comment:
Would this have correct behavior if there are multiple Offstage widgets present in the tree, with some onstage and some not?

Reply:
If you have a Navigator wrapped in an Offstage with offstage == false and everything wrapped in an Offstage with offstage == true this probably wouldn't do the right thing.